### ISSUE TITLE
Update normalize_path() to match wp_normalize_path() behavior

### ISSUE THREAD

**Issue by swissspidy** [wp-cli/wp-cli#5008]
[`normalize_path()`](https://github.com/wp-cli/wp-cli/blob/9559bb509450a9d2676e1985d21702fd4c2d73b5/php/utils.php#L787-L809) is basically a copy of the [`wp_normalize_path()`](https://developer.wordpress.org/reference/functions/wp_normalize_path/) core function.

In WordPress 4.9.7, that function was updated to allow for PHP file wrappers by using a [`wp_is_stream()`](https://developer.wordpress.org/reference/functions/wp_is_stream/) check.

To match core behavior we should think about adding a similar `is_stream()` utility function and extending `normalize_path()` with it.

**Comment by swissspidy** [comment]
Note: This was introduced in https://core.trac.wordpress.org/ticket/6821 / https://core.trac.wordpress.org/changeset/22094

`in_array()` is not enough I guess because the regex also checks for `://`

**Comment by schlessera** [comment]
The code first checks for the presence of `://` and bails early.

From there on, you could just substring the part before the `://` and do an `in_array()` instead. Faster and more portable.

**Comment by swissspidy** [comment]
Yeah that's true. I'm sure we can submit a core ticket for this :-)

**Comment by schlessera** [comment]
Here's my suggested version:
```php
function wp_is_stream( $path ) {
    $scheme_separator = strpos( $path, '://' );

    if ( false === $scheme_separator ) {
        // $path isn't a stream
        return false;
    }
 
    $stream = substr( $path, 0, $scheme_separator );

    return in_array( $stream, stream_get_wrappers(), true );
}
```

See https://3v4l.org/BTHk0
